### PR TITLE
Various updates for gcc 7.2.0

### PR DESCRIPTION
**Pull request by crystalct**

### CODE REVIEW

**Review by wargio — CHANGES_REQUESTED** [review]
Please delete the self that you have added.
```
samples/fireworks3D/fireworks3D.fake.self
```

**Comment by crystalct** [comment]
Done

File: lib/Makefile [hunk 1]
```diff
@@ -55,9 +79,8 @@ $(BUILD):
 
 shader:
 	@echo "[VPCOMP] $(notdir $<)"
-	@$(VPCOMP) shader/vshader_text_normal.vcg shader/vshader_text_normal.rvp
-	@$(RAW2H)  shader/vshader_text_normal.rvp source/vshader_text_normal.vcg.h source/vshader_text_normal.vcg.S vshader_text_normal_bin
-
+	@vpcomp shader/vshader_text_normal.vcg shader/vshader_text_normal.rvp
+	@raw2h.exe  shader/vshader_text_normal.rvp source/vshader_text_normal.vcg.h source/vshader_text_normal.vcg.S vshader_text_normal_bin
```
Comment (bucanero) [inline lib/Makefile:83]: ⚠️  this change using `raw2h.exe` directly won't work in systems that are not Windows

It's better to keep `@$(VPCOMP)` and `@$(RAW2H)` , so the user can define an export to the local tool independently of the operating system. For example:
```
export RAW2H=/my/path/to/raw2h
```

File: libfont/Makefile [hunk 2]
```diff
@@ -42,20 +68,23 @@ export BINFILES	:=	$(BINFILES:.bin=.bin.h)
...
+.PHONY: $(BUILD) install clean shader
 
 $(BUILD):
 	@[ -d $@ ] || mkdir -p $@
 	@make --no-print-directory -C $(BUILD) -f $(CURDIR)/Makefile
 
+shader:
+	@echo "[VPCOMP] $(notdir $<)"
+	@vpcomp shader/vshader_text_normal.vcg shader/vshader_text_normal.rvp
+	@raw2h.exe  shader/vshader_text_normal.rvp source/vshader_text_normal.vcg.h source/vshader_text_normal.vcg.S vshader_text_normal_bin
```
Comment (bucanero) [inline libfont/Makefile:83]: same `raw2h.exe` comment

**Review by bucanero — COMMENTED** [review]
Comments on `raw2h.exe` and Makefiles

File: lib/Makefile (hunk 1 above)
Comment (wargio) [reply lib/Makefile:83]: Fixed with 7e4080cf2d40b4c5b02d3c996a457fbcdad7a8fe

File: libfont/Makefile (hunk 2 above)
Comment (wargio) [reply libfont/Makefile:83]: Fixed with 7e4080cf2d40b4c5b02d3c996a457fbcdad7a8fe

**Pull request merged**